### DISCUSSION TITLE
How does NTP update work in this case?

### DISCUSSION THREAD

**Discussion by paolomeraviglia** [arendst/Tasmota#21341] [Q&A] [answered]
Hi there,

I am building up a project where an esp8266 will be sporadically connected to internet.
It will work pretty much standalone, and would be connected for 30-60 seconds randomly over wifi to internet every 1 - 2 days.
I need to use timers on that.

In that case, how NTP time update will work? Is It automatically run as soon as the device is connected to a wifi network? Is it possible to force that with a rule triggered by a wifi#connected event?

I am not concerned about time precision, because the time will be updated every 1-2 days, and my "execution window" has a buffer of 15 minutes

**Comment by sfromis** [comment, marked as answer]
As you should assume that the Tasmota device can be rebooted, you do not get reliable timers in that scenario. In some situations Tasmota may restart during attempts to get on Wifi again.

The general setup is that Tasmota will refresh NTP hourly which will not be triggered by a brief Wifi connection, but you could execute the command `Time 0` to trigger a synchronization.

A viable strategy could be to connect a RTC or GPS chip to allow Tasmota to have a reliable time source across reboots.
